tests/test_repository_pypi.py

```diff
@@ -347,3 +347,25 @@ def mock_get(*args, **kwargs):
 
     actual_data = pypi_repository._get_project(ireq)
     assert actual_data is None
+
+
+def test_name_collision(from_line, pypi_repository):
+    packages_base = os.path.join(os.path.dirname(__file__), "test_data", "packages")
+
+    name_collision_1 = (
+        "file://"
+        + str(os.path.join(packages_base, "fake_name_collision_1", "master.zip"))
```

Comment:
#1119 introducing new fixtures to create test packages rather than add binary archives to the VCS. I hope soon we refactor `test_data` in favor of functional fixtures to make tests more clear since it's difficult to maintain and review binary files. What do you think @vphilippon @AndydeCleyre?

Reply:
Agreed that this would be nicer if it weren't a couple of zips. Let me know if there's a better way to do it that fits with your general approach.

I explicitly wanted to avoid:
1. External dependencies.
2. Shelling out (or similar) to create the zips.

Reply:
#1119 is merged, so you can use the fixtures `make_package` and `make_sdist` now.

Reply:
OK, I've shifted to using those, and confirmed the test will fail without the new code.

Reply:
Looks good!